bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/GatewayWebTargets.java

```diff
@@ -0,0 +1,523 @@
+         * using their Generation 3 REST API to connect to their gateways. So we are asked to register with the gateway
+         * on startup. => So do not change the 'openhab.org' tag below !!
+         */
+        register = home + "integration/openhab.org";
+    }
+
+    /**
+     * Issue a command a activate a scene.
+     *
+     * @param sceneId the scene to be activated.
+     * @throws HubProcessingException if any error occurs.
+     */
+    public void activateScene(int sceneId) throws HubProcessingException {
+        invoke(HttpMethod.PUT, String.format(sceneActivate, sceneId), null, null);
+    }
+
+    /**
+     * Issue a calibrate command to a shade.
+     *
+     * @param shadeId the shade to be calibrated.
+     * @throws HubProcessingException if any error occurs.
+     */
+    public void calibrateShade(int shadeId) throws HubProcessingException {
+        String json = jsonParser.toJson(new ShadeMotion(ShadeMotion.Type.CALIBRATE));
+        invoke(HttpMethod.PUT, String.format(shadeMotion, shadeId), null, json);
+    }
+
+    @Override
+    public void close() throws IOException {
+        closing = true;
+        sseClose();
+    }
+
+    /**
+     * Register the binding with the hub (if not already registered).
+     *
+     * @throws HubProcessingException if any error occurs.
+     */
+    public boolean gatewayRegister() throws HubProcessingException {
+        if (!isRegistered) {
+            invoke(HttpMethod.PUT, register, null, null);
+            isRegistered = true;
+        }
+        return isRegistered;
+    }
+
+    /**
+     * Get hub properties.
+     *
+     * @return a map containing the hub properties.
+     * @throws HubProcessingException if any error occurs.
+     */
+    public Map<String, String> getInformation() throws HubProcessingException {
+        String json = invoke(HttpMethod.GET, info, null, null);
+        try {
+            Info result = jsonParser.fromJson(json, Info.class);
+            if (result == null) {
+                throw new HubProcessingException("getInformation(): missing response");
+            }
+            return Map.of( //
+                    Thing.PROPERTY_FIRMWARE_VERSION, result.getFwVersion(), //
+                    Thing.PROPERTY_SERIAL_NUMBER, result.getSerialNumber());
+        } catch (JsonParseException e) {
+            throw new HubProcessingException("getFirmwareVersions(): JsonParseException");
+        }
+    }
+
+    /**
+     * Get the list of scenes.
+     *
+     * @return the list of scenes.
+     * @throws HubProcessingException if any error occurs.
+     */
+    public List<Scene> getScenes() throws HubProcessingException {
+        String json = invoke(HttpMethod.GET, scenes, null, null);
+        try {
+            return List.of(jsonParser.fromJson(json, Scene[].class));
+        } catch (NullPointerException e) {
+            throw new HubProcessingException("getScenes() missing response");
```

Comment:
You should not catch `NullPointerException`, but instead avoid it.

Reply:
I think you need to address your comment to the author of the Java List class .. below is the JavaDoc for the `List.of()` method..
```
   /**
     * Returns an unmodifiable list containing an arbitrary number of elements.
     * See <a href="#unmodifiable">Unmodifiable Lists</a> for details.
     *
     * @apiNote
     * This method also accepts a single array as an argument. The element type of
     * the resulting list will be the component type of the array, and the size of
     * the list will be equal to the length of the array. To create a list with
     * a single element that is an array, do the following:
     *
     * <pre>{@code
     *     String[] array = ... ;
     *     List<String[]> list = List.<String[]>of(array);
     * }</pre>
     *
     * This will cause the {@link List#of(Object) List.of(E)} method
     * to be invoked instead.
     *
     * @param <E> the {@code List}'s element type
     * @param elements the elements to be contained in the list
     * @return a {@code List} containing the specified elements
     * @throws NullPointerException if an element is {@code null} or if the array is {@code null}
     *
     * @since 9
     */
    @SafeVarargs
    @SuppressWarnings("varargs")
    static <E> List<E> of(E... elements) {
```

Reply:
That documents exactly how you can avoid it, which is my point.